test/_data/prefer_asserts_in_initializer_list/lib.dart

```diff
@@ -0,0 +1,7 @@
+class A {
+  A.c1(a) : assert(a != null); // OK
+  A.c2(a) { // LINT
+    assert(a != null);
+  }
+  A.c3(a) {} // OK
+}
```

Comment:
My bad. Actually those comments are not important in integration tests, only the final result is tested https://github.com/dart-lang/linter/pull/677/files#diff-64cac337b62c744b4fa7ff04a8067aa0R444